### ISSUE TITLE
Track traffic to GitHub repo longer than 14 days

### ISSUE THREAD

**Issue by cirosantilli** [isaacs/github#399]
https://github.com/cirosantilli/test/graphs/traffic

Possible? If not, +1 from me.

On SE: http://webapps.stackexchange.com/questions/60915

**Comment by cirosantilli** [comment]
Ivan replied it is not currently possible.

**Comment by Lorenzo45** [comment]
+1

**Comment by greckyogurt** [comment]
+1

**Comment by Danappelxx** [comment]
:+1:

**Comment by alexbudmsft** [comment]
+1

**Comment by coni2k** [comment]
+1

**Comment by saqib-razzaq-aspose** [comment]
+1

**Comment by xXFracXx** [comment]
+1

**Comment by brianmc** [comment]
++1 for me

**Comment by nikhil-pandey** [comment]
:+1:

**Comment by sanyamc-msft** [comment]
+1 there are many people in my team who dearly want this data. Currently we manually parse the data into a csv and power our charts on a forthnightly basis. Would really appreciate api/ json data, even if its for 14 days.

**Comment by saqib-razzaq-aspose** [comment]
Sent from Outlook Mobile<https://aka.ms/blhgte>



On Fri, Feb 19, 2016 at 6:13 PM -0800, "sanyamc-msft" <notifications@github.com<mailto:notifications@github.com>> wrote:


+1 there are many people in my team who dearly want this data. Currently we manually parse the data into a csv and power our charts on a forthnightly basis. Would really appreciate api/ json data, even if its for 14 days.

**Comment by xianyi** [comment]
+1

**Comment by lolbot-iichan** [comment]
+1

**Comment by george-haddad** [comment]
+1

**Comment by nakulpathak3** [comment]
+1

**Comment by wvangeit** [comment]
+1

**Comment by rahulkrvnit** [comment]
really need this data

**Comment by zonky2** [comment]
+1

**Comment by rljordan** [comment]
+1

**Comment by ruhbehka** [comment]
+1

**Comment by KindKitty** [comment]
+1

**Comment by JarekToro** [comment]
+1

**Comment by ManoMarks** [comment]
+1

**Comment by johngronberg-okta** [comment]
+1

**Comment by ZahediAquino** [comment]
+1

**Comment by KrisSiegel** [comment]
This would be incredibly helpful especially for those maintaining open source libraries.

**Comment by mik3cap** [comment]
This is crazy, at least let the time period be 28 days!

**Comment by Chiel92** [comment]
+1

**Comment by burnash** [comment]
I made a little [web app](https://repotraffic.com/) that stores and plots traffic data for longer than 14 days. I'll be glad if someone finds it useful.

**Comment by jcfr** [comment]
@burnash Thanks for sharing, that is neat :+1: 

Out of curiosity, do you plan to release the associated source code ? I was only able to find https://github.com/repotraffic

Cc: @fedorov @thewtex

**Comment by evaame** [comment]
+1

**Comment by dikiaap** [comment]
+1

**Comment by Smittey** [comment]
+1
It seems like a bizarre restriction that other services like BitBucket do not have. It would be very useful

**Comment by tugrul512bit** [comment]
+1

I want to know total unique visitors since the beginning. Why not with a backtrack range such as t=14 or t=99999?

**Comment by guilherme6191** [comment]
+1

**Comment by pditommaso** [comment]
+1

**Comment by faridcher** [comment]
+1

**Comment by uekerman** [comment]
+1

**Comment by ohld** [comment]
+1

**Comment by sangallidavide** [comment]
+1

**Comment by karlvirgil** [comment]
+1

**Comment by dan-da** [comment]
+1.    seriously github?

**Comment by lneuhaus** [comment]
+1

**Comment by dsandip** [comment]
+1

**Comment by vinniefalco** [comment]
Obviously, its possible. And I am sure that they have this data for themselves.

**Comment by fedorov** [comment]
I found this package to be very helpful in setting up archival for github traffic collection: https://github.com/doowb/github-traffic

**Comment by captainswain** [comment]
+1

**Comment by IssacKramer** [comment]
+1

**Comment by Yaroslav-Lyutvinskiy** [comment]
+1

**Comment by KaiKostack** [comment]
+1

**Comment by M165437** [comment]
Just released a Laravel application that automatically fetches and persists repository traffic (visitors for now) on an daily basis and plots the data onto a chart. https://github.com/M165437/gh-traffic

**Comment by Yaroslav-Lyutvinskiy** [comment]
Thank you for perfect project, Michael!
Could I ask you, if you have it somewhere installed? I not very fond to php/apache stack and would find it rather difficult to instantiate it somewher. May be we can use your instance somehow?

**Comment by M165437** [comment]
@Yaroslav-Lyutvinskiy Have a look at https://gh-traffic.com ;-)

**Comment by reshamas** [comment]
@M165437 I tried following instructions for your repo and ran into errors.  Also, I'm only familiar with Python (not really PHP or Composer)

GitHub - can you PRETTY PLEASE add this feature?  It is in high demand, most valuable to us users.  It seems like if the data is available for past 2 weeks, it could be available for a longer period of time.  It would be greatly appreciated! 🙏

**Comment by les** [comment]
+1

**Comment by David-Levinthal** [comment]
+1

**Comment by platelminto** [comment]
+1

**Comment by gfoss** [comment]
+1

**Comment by sjlnk** [comment]
+1

**Comment by talbertc-usgs** [comment]
+1
Wait, can I make that +2?

**Comment by Trek333** [comment]
+1

**Comment by pragasitcha** [comment]
+1

**Comment by lacabra** [comment]
+1

**Comment by mdouze** [comment]
+1

**Comment by zagovora** [comment]
+1

**Comment by jpreiss** [comment]
+1

People contribute open source because they want to help people, so it's gratifying to see a count of those people.

**Comment by jdgarciauc3m** [comment]
+1

**Comment by nisargrthakkar** [comment]
+1

**Comment by jerryzhujian9** [comment]
+1

**Comment by tarun27sh** [comment]
+1

**Comment by Zamirk** [comment]
+1

**Comment by sunb0002** [comment]
+1

**Comment by fran-pena** [comment]
+1

**Comment by Henry00IS** [comment]
+1

**Comment by evejennings** [comment]
+1

**Comment by payton** [comment]
+1

**Comment by maartenvds** [comment]
+1

**Comment by Adam-Langley** [comment]
+1

**Comment by reshamas** [comment]
Can someone at GitHub confirm they've seen this suggestion to track traffic > 14 days?

**Comment by fedorov** [comment]
@reshamas this is not an official GitHub issue tracker. It's basically a place for GitHub users to find solace that they are not alone being frustrated about certain GitHub issue.

You might want to read the [README file of this repo](https://github.com/isaacs/github/blob/master/README.md), and make sure you added your voice by emailing GitHub support directly, as discussed in that README file:
> Post an issue if you have an issue or feature request for GitHub. **But you should also email support@github.com, since this repo is strictly for our own (unofficial) tracking purposes.** If they reply, (and they usually do, quickly) and if it is not a confidential matter like a security disclosure, add their reply to the issue so that other users can know what their official response was.

**Comment by abegehr** [comment]
+1

**Comment by lwhite1** [comment]
+1

**Comment by arjunhassard** [comment]
+1

**Comment by josecebe** [comment]
+288

**Comment by DavidJacobson** [comment]
+1

**Comment by djl11** [comment]
+42

**Comment by ic0ns** [comment]
+1

**Comment by butok** [comment]
+1

**Comment by geniusdibya** [comment]
> I made a little [web app](https://repotraffic.com/) that stores and plots traffic data for longer than 14 days. I'll be glad if someone finds it useful.
@burnash This is cool. But doesn't have the Clone data. Can you add that too?
Also is the code available on GIT to contribute?

**Comment by reshamas** [comment]
This is cool!

**Comment by timgrossmann** [comment]
@geniusdibya I'd really like to try your tool, but allowing full read and write access to all my public repositories only to view my traffic history seems a bit too much, doesn't it?
Why do you need write access?

**Comment by geniusdibya** [comment]
@timgrossmann I am not the author of repotraffic tool. Mine is --> [Pepipost](https://github.com/pepipost/), a cloud based email delivery platform.
Actually, @burnash is the author of repotraffic. I just commented by quoting his reply :)

**Comment by burnash** [comment]
@geniusdibya I love this thread :) 

Also, thank you for the feeback.

> But doesn't have the Clone data. Can you add that too?

Maybe. It'd be a nice feature to have. And it's has been already requested (https://github.com/repotraffic/repotraffic-feedback/issues/3 and https://github.com/repotraffic/repotraffic-feedback/issues/7)
I've been tracking my own open source project [gspread](https://github.com/burnash/gspread) for two years, but haven't really needed to track a number of clones.

> Also is the code available on GIT to contribute?

Not yet. But I'm considering open sourcing it (https://github.com/repotraffic/repotraffic-feedback/issues/6) if I have enough time.

**Comment by burnash** [comment]
@timgrossmann 

> I'd really like to try your tool, but allowing full read and write access to all my public repositories only to view my traffic history seems a bit too much, doesn't it?
> Why do you need write access?

Good question. *In short:* there's no way known to me to get the traffic data without the `public_repo` scope in an OAuth app. [GitHub asks for read/write permission to access traffic data](https://developer.github.com/v3/repos/traffic/#views):

> For repositories that you have push access to, the traffic API provides access to the information provided in the graphs section.

*Full answer:* https://github.com/repotraffic/repotraffic-feedback/issues/6#issuecomment-432638356

----
For those new to the thread: [Repotraffic](https://repotraffic.com/) is an app that stores and plots traffic data for longer than 14 days.

**Comment by timgrossmann** [comment]
@burnash Thank you for the answer and clarifying this.

That's pretty odd of instagram to ask for this. 
I'm not really into allowing any platform to access my all my projects with owner right permissions... 
Would've loved to use your tool.

Maybe at one point there will be a solution.

Thanks

**Comment by abbyachau** [comment]
+1 for the original request

**Comment by manojkumarvohra** [comment]
+1

**Comment by glenn-jocher** [comment]
> I made a little [web app](https://repotraffic.com/) that stores and plots traffic data for longer than 14 days. I'll be glad if someone finds it useful.
> @burnash This is cool. But doesn't have the Clone data. Can you add that too?
> Also is the code available on GIT to contribute?

This works great!

**Comment by timqian** [comment]
Hi all, I made another open-source [web app](https://repo-analytics.github.io/) recording repo traffics, clones, referrers and more.

- Website: [Repo Analytics](https://repo-analytics.github.io/)
- Source code: [GitHub](https://github.com/repo-analytics/repo-analytics.github.io)

**Comment by timgrossmann** [comment]
@timqian Same problem as with the other tooling... It's a shame that GitHub doesn't change that...

<img width="479" alt="Screenshot 2019-10-14 at 09 25 59" src="https://user-images.githubusercontent.com/16529337/66734762-a5f5cb00-ee64-11e9-9ad0-e9aae02f4634.png">

**Comment by timqian** [comment]
@timgrossmann Yeah, unfortunately, this is required 😂 Only thing I can do is to take care of your token. Also when you don't need this service, you can delete the token from your setting anytime

**Comment by UbhiTS** [comment]
it really is a shame that github has to make users create solutions just to know the extended history and analytics of their repositories :( .. I would prefer not to give anyone write access to my profile and repositories just to know extended analytics. Github, hope you're listening :)

**Comment by BradleyA** [comment]
I wrote a couple of scripts that you can find: https://github.com/BradleyA/Linux-admin/tree/master/github-repository-traffic#github-repository-traffic------------- It is a very simple shell script that uses crontab to schedule the download of GitHub traffic statistics in plain text from your GitHub repositories. Another script gives you the text lines to add to crontab and sets up the directories for GitHub traffic statistics. It takes minutes to setup but weeks to get enough data.  At this point you can process the data with any tools you wish.  I created a third script that parses the GitHub traffic and creates two markdown tables (clone, Views).

It solves my challenge. It may help you.

**Comment by MBurtsev** [comment]
Great and useful idea, I am sure that it should be implemented.

**Comment by UbhiTS** [comment]
this is seriously frustrating, in this age of cheap storage, GitHub would not even have a totals counter if not the entire history graph. At least give us Total Hits, Hits in last 1 year and hits in last 1 month. I'm sure having running counters would not add much to you load. For the love of god GitHub, please listen !

**Comment by bzhu5** [comment]
+1

**Comment by sangonzal** [comment]
For anyone still looking for a solution for this, I made a GitHub action that you might find useful: https://github.com/marketplace/actions/repository-traffic

With a GitHub action, the access token that is needed for accessing the GitHub API stays in your control.

**Comment by kevinmgates** [comment]
> For anyone still looking for a solution for this, I made a GitHub action that you might find useful: https://github.com/marketplace/actions/repository-traffic
> 
> With a GitHub action, the access token that is needed for accessing the GitHub API stays in your control.

AWESOME!

**Comment by matthansen0** [comment]
I wrote an automated solution for this also that uses Azure tech - https://github.com/matthansen0/logicapp-githubstats

**Comment by UbhiTS** [comment]
After my frustrations, I also wrote a solution for myself (and others) who do not wish to share any info with anyone else. No access token, no nothing. I basically trick github and found an easy way to maintain a running counter. 

https://medium.com/@ubhits/how-i-beat-githubs-14-day-only-traffic-insights-using-aws-lambda-sqs-dynamodb-and-the-api-98f9216dbe4c

**Comment by InbarGazit** [comment]
+1 for this feature

**Comment by antonkomarev** [comment]
I've made a pet project [Ÿ HŸPE](https://yhype.me) which helps me to collect and display GitHub traffic data longer than 14 days and even more.

![Ÿ HŸPE repo traffic](https://user-images.githubusercontent.com/1849174/112517492-6e08ec00-8da9-11eb-9868-d2222db4c88d.png)

**Comment by InbarGazit** [comment]
does it work retroactively? or only from that point forward?

**Comment by antonkomarev** [comment]
It can only start collecting data after installation -14 days. GitHub doesn't provide data from the past periods... :sob:

**Comment by glenn-jocher** [comment]
@antonkomarev do you know any way (paid or otherwise) that allows for more comprehensive github analytics (more similar to google analytics)?

One major limitation I see with the short 14 day numbers is that it's impossible to determine the number of unique visitors over an extended time period. For example if a repo has 1000 unique visitors over 2 weeks, this does not mean that there will be 2000 unique visitors over 4 weeks, or 4000 unique visitors over 8 weeks, etc, as many visitors over the following time period will be repeat visitors. Does my question make sense?

**Comment by glenn-jocher** [comment]
@antonkomarev @timqian do you know any way (paid or otherwise) that allows for more comprehensive github analytics over extended timespans (more similar to google analytics)?

One major limitation I see with accumulating the short 14 day numbers is that it's impossible to determine the number of unique visitors over an extended time period. For example if a repo has 1000 unique visitors over 2 weeks, this does not mean that there will be 2000 unique visitors over 4 weeks, or 4000 unique visitors over 8 weeks, etc, as many visitors over the following time period will be repeat visitors. Does my question make sense?

**Comment by antonkomarev** [comment]
@glenn-jocher in https://yhype.me we summarize the daily counters of unique visitors and if there was 3 unique visitors in January and 2 in March we tend to think that it was 5 different users.

GitHub says that it is unique visitors, but there is no 100% proof that if someone will be counted as unique visitor in January will not be treated as unique visitor in March one more time.

**Comment by jgehrcke** [comment]
I did have a rather close look at all existing solutions to 'work around this', for custom aggregation. I've built https://github.com/marketplace/actions/github-repo-stats / https://github.com/jgehrcke/github-repo-stats and you might want to have  a look! It generates a nice HTML and PDF report and from my point of view the most important part is that the data stays in GitHub. The time series data is simply stored in a "data repository", with transparent evolution of history. The HTML report can be exposed via GitHub Pages. Example (demo): https://jgehrcke.github.io/ghrs-test/jgehrcke/covid-19-germany-gae/latest-report/report.html

Would appreciate feedback! Thanks!

**Comment by ElliotB256** [comment]
To add something to this discussion, GitHub _does_ store the access data for repositories for longer than the 14 day period. I recently filed a DMCA takedown on an infringing repository, and they were able to provide page views and clones going back a few months for the repo on request. I'm not sure why the data is not made easily available.
